projects/power_distribution/src/front_uv_detector.c

```diff
@@ -0,0 +1,30 @@
+#include "front_uv_detector.h"
+
+#include "can_transmit.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "pin_defs.h"
+
+static GpioAddress s_uv_comp_pin_address = FRONT_UV_COMPARATOR_PIN;
+
+static void prv_pin_interrupt_handler(const GpioAddress *address, void *context) {
+  CAN_TRANSMIT_UV_CUTOFF_NOTIFICATION();
+
+  LOG_WARN("UV lockout was triggered\n");
+}
+
+StatusCode front_uv_detector_init(GpioAddress *detector_pin) {
+  gpio_it_init();
```

Comment:
Initialize this in `setup_test` not here! In power distribution, main.c already initializes `gpio_it`, and initializing it again could interfere with other modules.